unit_testing/test_centerline.py

```diff
@@ -145,21 +152,20 @@ def test_get_centerline_nurbs(img_ctl, expected, params):
 
 
 # noinspection 801,PyShadowingNames
-def test_get_centerline_optic():
-    """Test extraction of metrics aggregation across slices: All slices by default"""
-    fname_t2 = os.path.join(__sct_dir__, 'sct_testing_data/t2/t2.nii.gz')  # install: sct_download_data -d sct_testing_data
-    img_t2 = Image(fname_t2)
+@pytest.mark.parametrize('params', param_optic)
+def test_get_centerline_optic(params):
+    """Test centerline extraction with optic"""
+    # TODO: add assert on the output .csv files for more precision
+    im = Image(params['fname_image'])
     # Add non-numerical values at the top corner of the image for testing purpose
-    img_t2.change_type('float32')
-    img_t2.data[0, 0, 0] = np.nan
-    img_t2.data[1, 0, 0] = np.inf
-    img_out, arr_out, _, _ = get_centerline(
-        img_t2, ParamCenterline(algo_fitting='optic', contrast='t2', minmax=False), verbose=VERBOSE)
+    im.change_type('float32')
+    im.data[0, 0, 0] = np.nan
+    im.data[1, 0, 0] = np.inf
+    im_centerline, arr_out, _, _ = get_centerline(
+        im, ParamCenterline(algo_fitting='optic', contrast='t2', minmax=False), verbose=VERBOSE)
     # Open ground truth segmentation and compare
-    fname_t2_seg = os.path.join(__sct_dir__, 'sct_testing_data/t2/t2_seg.nii.gz')
-    img_seg_out, arr_seg_out, _, _ = get_centerline(
-        Image(fname_t2_seg), ParamCenterline(algo_fitting='bspline', minmax=False), verbose=VERBOSE)
-    assert np.linalg.norm(find_and_sort_coord(img_seg_out) - find_and_sort_coord(img_out)) < 3.5
+    assert sct.math.dice(im_centerline.data,
+                         Image(params['fname_centerline-optic']).data) == 1.0
```

Comment:
What was the reason for replacing mse by dice here? more sensitive?

Reply:
> What was the reason for replacing mse by dice here? more sensitive?

much more sensitive: previously we were doing fitting to centerline from ground truth segmentation, then mse with arbitrary threshold (3.5). Now, we want the output centerline to be *exactly the same* as the ground truth